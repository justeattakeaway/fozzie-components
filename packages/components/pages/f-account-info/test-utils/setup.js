import Vuex from 'vuex';
import { createLocalVue } from '@vue/test-utils';
import { VueI18n } from '@justeat/f-globalisation';
import tenantConfigs from '../src/tenants';

export const localVue = createLocalVue();
localVue.use(VueI18n);
localVue.use(Vuex);

export const i18n = {
    locale: 'en-GB',
    messages: {
        'en-GB': tenantConfigs['en-GB'].messages
    }
};

export const baseUrl = 'https://smartGatewayBaseUrl.com';

export const token = 'some-auth-token';

export const conversationId = 'b7386108-95e6-4e73-9421-5b066c089153';

export const consumerDetailsGetResponse = {
    data: {
        ConsumerId: 1050450174,
        Id: '8It2lEaJjG82nJtxO00Iab0CoF0=',
        CreatedDate: '2019-04-16T17:25:00.000Z',
        EmailAddress: 'satchin.moriyoka@example.com',
        FirstName: 'Satchin',
        LastName: 'Moriyoka',
        PhoneNumber: '+441234567890',
        DateOfBirth: '1914-04-16T17:25:00.000Z',
        TermsAndConditionsId: 0,
        WantsNewsletter: true,
        AccountType: 'Default',
        ConsumerStatus: 'ReturnConsumer'
    }
};

const consumerAddressGetResponse = {
    AddressId: 1050450174,
    City: 'Bristol',
    ZipCode: 'BS1 4DJ',
    AddressName: 'Office',
    IsDefault: true, // Key field
    Line1: 'Just Eat',
    Line2: 'Broad Quay House'
};

export const consumerAddressesGetResponse = {
    data: [{
        ...consumerAddressGetResponse
    },
    {
        AddressId: 1050450175,
        City: 'Bristol',
        ZipCode: 'BS1 9DJ',
        AddressName: 'Office',
        IsDefault: false,
        Line1: 'Takeaway',
        Line2: 'Broad Quay House'
    }]
};

export const consumerDetailsMappedModel = {
    id: consumerDetailsGetResponse.data.ConsumerId,
    firstName: consumerDetailsGetResponse.data.FirstName,
    lastName: consumerDetailsGetResponse.data.LastName,
    phoneNumber: consumerDetailsGetResponse.data.PhoneNumber,
    emailAddress: consumerDetailsGetResponse.data.EmailAddress
};

export const consumerAddressMappedModel = {
    line1: consumerAddressGetResponse.Line1,
    line2: consumerAddressGetResponse.Line2,
    line3: consumerAddressGetResponse.Line3,
    locality: consumerAddressGetResponse.City,
    postcode: consumerAddressGetResponse.ZipCode
};

export const consumerViewModel = {
    ...consumerDetailsMappedModel,
    ...consumerAddressMappedModel
};

export const consumerStateModel = {
    consumer: {
        ...consumerDetailsMappedModel,
        ...consumerAddressMappedModel
    }
};
