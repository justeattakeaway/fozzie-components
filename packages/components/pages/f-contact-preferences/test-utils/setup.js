export const contactPreferencesGetResponse = {
    data: {
        Preferences: [{
            DisplayName: 'Order status',
            Sort: 1,
            Key: 'orderstatus',
            Push: false,
            Email: true,
            Sms: true
        }, {
            DisplayName: 'Review meal',
            Sort: 2,
            Key: 'reviewmeal',
            Push: false,
            Email: false,
            Sms: false
        }, {
            DisplayName: 'News & offers',
            Sort: 3,
            Key: 'news',
            Push: false,
            Email: false,
            Sms: false
        }
        ],
        PreferenceVersionViewed: 0
    }
};

export const contactPreferencesViewModel = {
    preferenceVersionViewed: 0,
    preferences: [{
        displayName: 'Order status',
        isEmailEnabled: false,
        emailValue: true,
        key: 'orderstatus',
        pushEnabled: false,
        pushValue: false,
        isSmsEnabled: false,
        smsValue: true,
        sort: 1,
        visible: true
    }, {
        displayName: 'Review meal',
        isEmailEnabled: true,
        emailValue: false,
        key: 'reviewmeal',
        pushEnabled: true,
        pushValue: false,
        isSmsEnabled: true,
        smsValue: false,
        sort: 2,
        visible: false
    }, {
        displayName: 'News & offers',
        isEmailEnabled: true,
        emailValue: false,
        key: 'news',
        pushEnabled: true,
        pushValue: false,
        isSmsEnabled: true,
        smsValue: false,
        sort: 3,
        visible: true
    }]
};

export const filteredContactPreferencesModel = [{
    displayName: 'Order status',
    isEmailEnabled: false,
    emailValue: true,
    key: 'orderstatus',
    pushEnabled: false,
    pushValue: false,
    isSmsEnabled: false,
    smsValue: true,
    sort: 1,
    visible: true
}, {
    displayName: 'News & offers',
    isEmailEnabled: true,
    emailValue: false,
    key: 'news',
    pushEnabled: true,
    pushValue: false,
    isSmsEnabled: true,
    smsValue: false,
    sort: 3,
    visible: true
}];

export const contactPreferencesUpdateModel = {
    Preferences: [{
        DisplayName: 'Order status',
        Sort: 1,
        Key: 'orderstatus',
        Push: false,
        Email: true,
        Sms: true
    }, {
        DisplayName: 'Review meal',
        Sort: 2,
        Key: 'reviewmeal',
        Push: false,
        Email: false,
        Sms: false
    }, {
        DisplayName: 'News & offers',
        Sort: 3,
        Key: 'news',
        Push: false,
        Email: false,
        Sms: false
    }],
    DeviceToken: null,
    DeviceType: null,
    PhoneNumber: null,
    PreferenceVersionViewed: 0
};

export const baseUrl = 'https://smartGatewayBaseUrl.com';

export const token = 'some-auth-token';

export const conversationId = 'b7386108-95e6-4e73-9421-5b066c089153';

export const contactPreferencesUpdateBody = {
    Preference: [{
        DisplayName: 'Order status',
        Sort: 1,
        Key: 'orderstatus',
        Push: false,
        Email: true,
        Sms: true
    }, {
        DisplayName: 'Review meal',
        Sort: 2,
        Key: 'reviewmeal',
        Push: false,
        Email: false,
        Sms: false
    }, {
        DisplayName: 'News & offers',
        Sort: 3,
        Key: 'news',
        Push: false,
        Email: true,
        Sms: false
    }
    ],
    DeviceToken: null,
    DeviceType: null,
    PhoneNumber: null,
    PreferenceVersionViewed: 0
};
