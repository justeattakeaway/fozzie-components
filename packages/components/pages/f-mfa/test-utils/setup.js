export const baseUrl = 'https://smartGatewayBaseUrl.com';

export const conversationId = 'b7386108-95e6-4e73-9421-5b066c089153';

// eslint-disable-next-line camelcase
export const postBody = { mfa_token: 'some_mfa_token', otp: 'ABC123' };
