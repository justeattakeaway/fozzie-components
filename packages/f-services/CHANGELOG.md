# Changelog

The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).


v1.2.0
------------------------------
*September 2, 2020*

### Changed
- Handle missing navigator API so that the default axios timeout is returned when used on the server.


v1.1.0
------------------------------
*August 26, 2020*

### Added
- Request/response interceptors to add response times to response object.
- `axios-mock-adapter` to help with testing.

### Changed
- Vue CLI minor package updates.
- `@vue/cli-plugin-unit-test` to v4.3.1.
- ESLint autofix turned off (so that tests don't pass due to `--fix` being applied, but then publish subsequently fails)


v1.0.1
------------------------------
*May 6, 2020*

### Changed
- Syntax in `index.js` so that newer versions of f-services are compatible with Storybook


v1.0.0
------------------------------
*April 1, 2020*

### Added
- Axios services module.
- Axios as a peer dependency.

### Changed
- Split central "shared" module into separate "window" and "globalisation" modules.
- `jest` config updates. `moduleNameMapper` removed, `jsx` removed from checks (as we don't use) and `transformIgnorePatterns` updated.
- Rollup package updates.

### Fixed
- `addEvent` returns the function that was assigned to the event listener so that this can be passed in to `removeEvent`. This is necessary for removing throttled event listeners.


v0.13.2
------------------------------
*February 24, 2020*

### Changed
- Unneeded dependencies removed (as in root monorepo dependencies or not used)


v0.13.1
------------------------------
*January 22, 2020*

### Changed
- Refactor JSDoc to conform to syntax rules
- Refactor `getLocale` function to reduce complexity
- Refactor `addEvent` function to reduce complexity

### Added
- `DEFAULT_LOCALE` variable for appropriate fallback rules


v0.13.0
------------------------------
*November 8, 2019*

### Changed
- Making sure that necessary 3rd party dependencies are bundled as part of the package


v0.12.0
------------------------------
*November 8, 2019*

### Changed
- Switched over to rollup for bundling as webpack was giving strange results when importing as a library


v0.11.0
------------------------------
*October 4, 2019*

### Added
- `window-or-global` package for SSR issue`


v0.10.0
------------------------------
*October 2, 2019*

### Added
- Tests for new services that have been added

### Changed
- `webpack.config.js` and packages for a better build
- Name of the export in the `index.js` to `sharedServices` for consistency


v0.9.0
------------------------------
*October 1, 2019*

### Added
- `lodash-es` throttle for the `addEvent` service


v0.8.0
------------------------------
*September 30, 2019*

### Added
- `getWindowWidth` method for returning the current width of the window
- `getWindowHeight` method for returning the current height of the window
- `addEvent` method for listening to a given event and applying a given function at a given time
- `removeEvent` method for clearing the listener for a given event and function


v0.7.0
------------------------------
*September 27, 2019*

### Removed
- `transformLocale` as it is not needed any more


v0.6.0
------------------------------
*September 26, 2019*

### Changed
- Webpack config to add babel polyfill


v0.5.0
------------------------------
*September 26, 2019*

### Added
- `transformLocale` method for changing the case format of the locale


v0.4.0
------------------------------
*September 25, 2019*

### Added
- String manipulation on locale in `getLocale` method


v0.3.0
------------------------------
*September 24, 2019*

### Added
- Added fallback for other method in shared service


v0.2.0
------------------------------
*September 24, 2019*

### Added
- Fallback for locale in shared logic


v0.1.0
------------------------------
*September 24, 2019*

### Added
- Created service package for shared logic
- Tests for the methods inside
