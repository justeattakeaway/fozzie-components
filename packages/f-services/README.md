## Usage

1.  Install the module using NPM or Yarn:

    ```bash
    yarn add @justeat/f-services
    ```

    ```bash
    npm install @justeat/f-services
    ```

2.  Import the package

    You can import it in your like this (please note that styles have to be imported separately)

    ```
    import VueServices from '@justeat/f-services';

    ```

    If you are using Webpack, you can import the component dynamically to separate the services bundle from the main `bundle.client.js`:

3. Services in the bundle:

    `getLocale` - Returns the locale where the user is logged in

    `getTheme` - Returns the theme based on the users locale

## Documentation to be completed once module is in stable state.
