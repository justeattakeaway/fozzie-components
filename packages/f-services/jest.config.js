module.exports = {
    moduleFileExtensions: [
        'js',
        'jsx',
        'json'
    ],

    transform: {
        '^.+\\.jsx?$': 'babel-jest'
    },

    transformIgnorePatterns: [
        'node_modules/(?!(babel-jest|jest-vue-preprocessor|lodash-es)/)'
    ],

    moduleNameMapper: {
        '^@/(.*)$': '<rootDir>/src/$1'
    },

    testMatch: [
        '**/*.{spec|test}.(js|jsx|ts|tsx)|**/__tests__/*.(js|jsx|ts|tsx)'
    ],

    testURL: 'http://localhost/'
};
