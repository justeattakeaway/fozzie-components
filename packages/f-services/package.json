{
  "name": "@justeat/f-services",
  "description": "Fozzie Services - Shared Services for Components and projects",
  "version": "1.0.0",
  "main": "dist/f-services.umd.js",
  "module": "dist/f-services.esm.js",
  "source": "src/index.js",
  "files": [
    "dist"
  ],
  "homepage": "https://github.com/justeat/fozzie-components/tree/master/packages/f-services",
  "contributors": [
    "Github contributors <https://github.com/justeat/fozzie-components/graphs/contributors>"
  ],
  "repository": {
    "type": "git",
    "url": "git@github.com:justeat/fozzie-components.git"
  },
  "bugs": {
    "url": "https://github.com/justeat/fozzie-components/issues"
  },
  "license": "Apache-2.0",
  "engines": {
    "node": ">=10.0.0"
  },
  "keywords": [
    "fozzie"
  ],
  "scripts": {
    "prepublishOnly": "yarn lint && yarn test && yarn build",
    "build": "rollup --config",
    "lint": "vue-cli-service lint",
    "lint:fix": "yarn lint --fix",
    "test": "vue-cli-service test:unit"
  },
  "browserslist": [
    "extends @justeat/browserslist-config-fozzie"
  ],
  "dependencies": {
    "lodash-es": "4.17.15",
    "window-or-global": "1.0.1"
  },
  "devDependencies": {
    "@rollup/plugin-commonjs": "11.0.2",
    "@rollup/plugin-json": "4.0.2",
    "@rollup/plugin-node-resolve": "7.1.1",
    "@vue/cli-plugin-eslint": "3.9.2",
    "@vue/cli-plugin-unit-jest": "4.0.5",
    "rollup": "1.26.3",
    "rollup-plugin-babel": "4.3.3",
    "rollup-plugin-terser": "5.1.2"
  },
  "peerDependencies": {
    "axios": "^0.19.2"
  }
}
