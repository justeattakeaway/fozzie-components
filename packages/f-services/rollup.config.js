import { terser } from 'rollup-plugin-terser';
import babel from 'rollup-plugin-babel';
import pkg from './package.json';

export default {
    input: pkg.source,
    output: [
        // commonjs (node)/es module config
        { file: pkg.main, format: 'cjs' },
        { file: pkg.module, format: 'es' },

        // browser package config
        {
            file: pkg.browser,
            format: 'umd',
            name: 'f-services'
        }
    ],
    plugins: [
        terser(),
        babel({
            exclude: ['node_modules/**']
        })
    ]
};
