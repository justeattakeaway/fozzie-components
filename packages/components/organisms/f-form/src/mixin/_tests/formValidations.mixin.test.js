import { shallowMount } from '@vue/test-utils';
import { validations } from '@justeat/f-services';
import {
    localVue,
    i18n,
    $v
} from '../../components/_tests/helpers';
import {
    formData,
    fieldWithoutValidations,
    fieldWithBothValidations,
    phoneNumberData,
    postcodeData
} from '../../../stories/helpers';

import formValidations from '../formValidation.mixin';

describe('Form', () => {
    let baseComponent;

    beforeEach(() => {
        baseComponent = {
            render () {},
            data () {
                return {
                    formData,
                    locale: 'en-GB'
                };
            },
            mixins: [formValidations]
        };
    });

    describe('computed :: ', () => {
        describe('validationState ::', () => {
            const expectedState = {
                invalidFields: [],
                validFields: []
            };
            let getFormValidationStateSpy;

            beforeEach(() => {
                getFormValidationStateSpy = jest.spyOn(validations, 'getFormValidationState').mockReturnValue(expectedState);
            });

            afterEach(() => {
                jest.clearAllMocks();
            });

            it('should call `getFormValidationState` with validations and return value', () => {
                const wrapper = shallowMount(baseComponent, {
                    i18n,
                    localVue,
                    mocks: {
                        $v: {
                            ...$v,
                            dirty: false
                        }
                    }
                });

                // Act
                const { validationState } = wrapper.vm;

                // Assert
                expect(getFormValidationStateSpy).toHaveBeenCalledWith($v);
                expect(validationState).toEqual(expectedState);
            });
        });

        describe('invalidFieldsSummary ::', () => {
            let wrapper;

            it('should return `null` if no fields have been touched', () => {
                wrapper = shallowMount(baseComponent, {
                    i18n,
                    localVue,
                    mocks: {
                        $v: {
                            ...$v,
                            dirty: false
                        }
                    }
                });

                // Assert
                expect(wrapper.vm.invalidFieldsSummary).toEqual(null);
            });

            describe('when fields are `dirty`', () => {
                let getFormValidationStateSpy;

                beforeEach(() => {
                    wrapper = shallowMount(baseComponent, {
                        i18n,
                        localVue,
                        mocks: {
                            $v: {
                                ...$v,
                                dirty: true
                            }
                        }
                    });

                    getFormValidationStateSpy = jest.spyOn(validations, 'getFormValidationState');
                });

                afterEach(() => {
                    jest.clearAllMocks();
                });

                it('should return `null` if all fields have been touched and are valid', () => {
                    // Arrange
                    const mockValidationState = {
                        invalidFields: []
                    };

                    getFormValidationStateSpy.mockReturnValue(mockValidationState);

                    // Assert
                    expect(wrapper.vm.invalidFieldsSummary).toEqual(null);
                });

                it('should return the error summary with the number of invalid fields', () => {
                    // Arrange
                    const mockValidationState = {
                        invalidFields: [
                            'mobileNumber',
                            'line1',
                            'postcode'
                        ]
                    };

                    getFormValidationStateSpy.mockReturnValue(mockValidationState);

                    wrapper = shallowMount(baseComponent, {
                        i18n,
                        localVue,
                        mocks: {
                            $v: {
                                formFields: {
                                    ...$v.formFields,
                                    $dirty: true
                                }
                            },
                            $tc: jest.fn(() => 'There are 3 errors in the form')
                        }
                    });

                    // Assert
                    expect(wrapper.vm.invalidFieldsSummary).toMatchSnapshot();
                });
            });
        });
    });

    describe('methods :: ', () => {
        afterEach(() => {
            jest.clearAllMocks();
        });

        describe('isFormValid ::', () => {
            afterEach(() => {
                jest.clearAllMocks();
            });

            it('should call `.touch()` ', () => {
                const wrapper = shallowMount(baseComponent, {
                    i18n,
                    localVue,
                    mocks: {
                        $v: {
                            ...$v,
                            dirty: false
                        }
                    }
                });

                wrapper.vm.isFormValid();

                // Assert
                expect($v.$touch).toBeCalled();
            });

            it.each([
                [true, false],
                [false, true]
            ])('should return %s if `$v.invalid` is $s', (expected, $invalid) => {
                const wrapper = shallowMount(baseComponent, {
                    i18n,
                    localVue,
                    mocks: {
                        $v: {
                            ...$v,
                            $invalid
                        }
                    }
                });

                // Assert
                expect(wrapper.vm.isFormValid()).toBe(expected);
            });
        });

        describe('isValidPhoneNumber ::', () => {
            afterEach(() => {
                jest.clearAllMocks();
            });

            it('should call `isValidPhoneNumber`', () => {
                // Arrange
                const isValidPhoneNumberSpy = jest.spyOn(validations, 'isValidPhoneNumber');

                const wrapper = shallowMount(baseComponent, {
                    i18n,
                    localVue
                });

                // Act
                wrapper.vm.isValidPhoneNumber();

                // Assert
                expect(isValidPhoneNumberSpy).toHaveBeenCalledWith(phoneNumberData.value, 'en-GB');
            });
        });

        describe('isValidPostcode ::', () => {
            afterEach(() => {
                jest.clearAllMocks();
            });

            it('should call `isValidPostcode`', () => {
                // Arrange
                const isValidPostcodeSpy = jest.spyOn(validations, 'isValidPostcode');

                const wrapper = shallowMount(baseComponent, {
                    i18n,
                    localVue
                });

                // Act
                wrapper.vm.isValidPostcode();

                // Assert
                expect(isValidPostcodeSpy).toHaveBeenCalledWith(postcodeData.value, 'en-GB');
            });
        });

        describe('fieldErrorStatus ::', () => {
            describe('when field does not have validations', () => {
                beforeEach(() => {
                    baseComponent.data = () => ({
                        formData: {
                            formFields: [
                                fieldWithoutValidations
                            ]
                        }
                    });
                });

                it('should return false', () => {
                    // Arrange
                    const wrapper = shallowMount(baseComponent, {
                        i18n,
                        localVue
                    });

                    // Act
                    const result = wrapper.vm.fieldErrorStatus(fieldWithoutValidations.name);

                    // Assert
                    expect(result).toEqual(false);
                });
            });

            it('should return false when field is not dirty', () => {
                // Arrange
                const wrapper = shallowMount(baseComponent, {
                    i18n,
                    localVue,
                    mocks: {
                        $v: {
                            formFields: {
                                fieldWithBothValidations: {
                                    $dirty: false
                                }
                            }
                        }
                    }
                });

                // Act
                const result = wrapper.vm.fieldErrorStatus(fieldWithBothValidations.name);

                // Assert
                expect(result).toEqual(false);
            });

            describe('when field is dirty and has validations', () => {
                beforeEach(() => {
                    baseComponent.data = () => ({
                        formData: {
                            formFields: [
                                fieldWithBothValidations
                            ]
                        }
                    });
                });

                it.each([
                    [false, true, true],
                    ['required', false, false],
                    ['required', false, true],
                    ['invalid', true, false]
                ])('should return %s if required is %s and isValid is %s', (expected, required, isValid) => {
                    // Arrange
                    const wrapper = shallowMount(baseComponent, {
                        i18n,
                        localVue,
                        mocks: {
                            $v: {
                                formFields: {
                                    email: {
                                        required,
                                        isValid,
                                        $dirty: true
                                    }
                                }
                            }
                        }
                    });

                    // Act
                    const result = wrapper.vm.fieldErrorStatus(fieldWithBothValidations.name);

                    // Assert
                    expect(result).toEqual(expected);
                });
            });
        });
    });
});
