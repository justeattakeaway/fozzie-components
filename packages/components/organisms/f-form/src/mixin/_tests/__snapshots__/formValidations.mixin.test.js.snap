// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form computed ::  invalidFieldsSummary :: when fields are \`dirty\` should return the error summary with the number of invalid fields 1`] = `"There are 3 errors in the form"`;
